IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/wolfssl_demo.c

```diff
@@ -294,78 +336,129 @@ static void Tls_client()
-        tsip_set_callback_ctx(ssl, &userContext);
+         #if !defined(TLS_MULTITHREAD_TEST)
+          memset(&userContext, 0, sizeof(TsipUserCtx));
+          tsip_set_callback_ctx(ssl, &userContext);
+         #else
+          if (p->port - TLSSERVER_PORT == 0) {
+             memset(&userContext_taskA, 0, sizeof(TsipUserCtx));
+             userContext_taskA.devId = 1;
+             tsip_set_callback_ctx(ssl, (void*)&userContext_taskA);
+          }
+          else {
+             memset(&userContext_taskB, 0, sizeof(TsipUserCtx));
+             userContext_taskB.devId = 2;
+             tsip_set_callback_ctx(ssl, (void*)&userContext_taskB);
+          }
+         #endif
         #endif
     }
 
+    msg(pcName, p->id, "  Cipher : %s\n", p->cipher);
+    /* use specific cipher */
+    if (p->cipher != NULL &&
+        wolfSSL_set_cipher_list(ssl, p->cipher) != WOLFSSL_SUCCESS) {
+        ret = -1;
+    }
+
     if (ret == 0) {
         /* associate socket with ssl object */
         if (wolfSSL_set_fd(ssl, (int)socket) != WOLFSSL_SUCCESS) {
-            printf("ERROR wolfSSL_set_fd: %d\n", wolfSSL_get_error(ssl, 0));
+            msg(pcName, p->id, "ERROR wolfSSL_set_fd: %d\n", 
+                                                    wolfSSL_get_error(ssl, 0));
             ret = -1;
         }
     }
-
+#ifdef DEBUG_WOLFSSL
+    wolfSSL_Debugging_ON();
+#endif
     if (ret == 0) {
         if (wolfSSL_connect(ssl) != WOLFSSL_SUCCESS) {
-            printf("ERROR wolfSSL_connect: %d\n", wolfSSL_get_error(ssl, 0));
+            msg(pcName, p->id, "ERROR wolfSSL_connect: %d\n", 
+                                                    wolfSSL_get_error(ssl, 0));
             ret = -1;
         }
     }
-
+#ifdef DEBUG_WOLFSSL
+   wolfSSL_Debugging_OFF();
+#endif
     if (ret == 0) {
         if (wolfSSL_write(ssl, sendBuff, strlen(sendBuff)) != 
                                                             strlen(sendBuff)) {
-            printf("ERROR wolfSSL_write: %d\n", wolfSSL_get_error(ssl, 0));
+            msg(pcName, p->id, "ERROR wolfSSL_write: %d\n", 
+                                                    wolfSSL_get_error(ssl, 0));
             ret = -1;
         }
     }
 
     if (ret == 0) {
         if ((ret=wolfSSL_read(ssl, rcvBuff, BUFF_SIZE -1)) < 0) {
-            printf("ERROR wolfSSL_read: %d\n", wolfSSL_get_error(ssl, 0));
+            msg(pcName, p->id, "ERROR wolfSSL_read: %d\n", 
+                                                    wolfSSL_get_error(ssl, 0));
             ret = -1;
         }
         else {
             rcvBuff[ret] = '\0';
-            printf("Received: %s\n\n", rcvBuff);
+            msg(pcName, p->id, "Received: %s\n\n", rcvBuff);
             ret = 0;
         }
     }
-
     
-    wolfSSL_shutdown(ssl);
-
-    FreeRTOS_shutdown(socket, FREERTOS_SHUT_RDWR);
-    while(FreeRTOS_recv(socket, rcvBuff, BUFF_SIZE -1, 0) >=0) {
-    	vTaskDelay(250);
+out:
+    if (ssl) {
```

Comment:
Was variable `ssl` initialized to NULL before coming here?

Reply:
There is there is a possibility of becoming a NULL when `ssl = wolfSSL_new(ctx);` is failure somehow.

Reply:
Variable `ssl` declared in the begging of this function is not initialized to NULL. Is it possible to come here if FreeRTOS_connect returns other than zero?  If possible, wolfSSL_shutdown or wolfSSL_free would give harm.